openhtf/core/test_executor.py

```diff
@@ -146,6 +146,12 @@ def _thread_proc(self):
 
       # Everything is set, set status and begin test execution.
       self._execute_test_phases(phase_exec)
+      
+  def _thread_exception(self, exc_type, exc_val, exc_tb):
+    """When the framework throws an exception, turn it into a failure code."""
```

Comment:
I think we should clarify `When the framework throws an exception` a little more. Looks like this is called when an exception is raised inside a test but outside of all phases?